### PR TITLE
added compressor reading

### PR DESCRIPTION
**Pull request by tharunS123**

### CODE REVIEW

File: src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.Compressor;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import team.gif.robot.RobotMap;
+
+public class RobotCompressor extends Compressor {
+    public static AnalogInput pressureSensor = null;
+    public RobotCompressor(int port, PneumaticsModuleType moduleType) {
+        super(port, moduleType);
+        pressureSensor = new AnalogInput(RobotMap.SENSOR_AIR_PRESSURE);
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java]: I know you copied this from previous years, but we should change RobotMap.SENSOR_AIR_PRESSURE to an argument.
If NULL, then don't instantiate the pressure sensor

File: src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package team.gif.robot.subsystems.drivers;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.Compressor;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import team.gif.robot.RobotMap;
+
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java:6]: We should include a description of the arguments here

File: src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+public class RobotCompressor extends Compressor {
+    public static AnalogInput pressureSensor = null;
+    public RobotCompressor(int port, PneumaticsModuleType moduleType) {
+        super(port, moduleType);
+        pressureSensor = new AnalogInput(RobotMap.SENSOR_AIR_PRESSURE);
+    }
+
+    public double getPressure() {
+        return 250 * (pressureSensor.getVoltage() / 4.82) - 25;
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java]: add if pressureSensor == NULL return -1

**Review by billknopfjr — CHANGES_REQUESTED** [review]

File: src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java (hunk 1 above)
Comment (tharunS123) [reply src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java]: So, the RobotMap.SENSOR_AIR_PRESSURE should be changed to NULL, right?

Comment (billknopfjr) [reply src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java]: No ...
```suggestion
public RobotCompressor(int port, PneumaticsModuleType moduleType, int sensorID) {
        pressureSensor = new AnalogInput(sensorID);
```

and in Robot.java

compressor = new RobotCompressor(RobotMap.COMPRESSOR, PneumaticsModuleType.CTREPCM, RobotMap.SENSOR_AIR_PRESSURE);

File: src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java (hunk 3 above)
Comment (billknopfjr) [reply src/main/java/team/gif/robot/subsystems/drivers/RobotCompressor.java]:
```suggestion
   if( pressureSensor )
        return 250 * (pressureSensor.getVoltage() / 4.82) - 25;
   else
        return -1;
```

**Pull request closed without merging**